src/ParallelAlgorithms/Interpolation/Protocols/PostInterpolationCallback.hpp

```diff
@@ -116,11 +117,25 @@ struct has_signature_3<
 struct PostInterpolationCallback {
   template <typename ConformingType>
   struct test {
-    using observation_types = typename ConformingType::observation_types;
+    static_assert(
+        std::is_same_v<ConformingType, void>,
+        "PostInterpolationCallback conforming type must be templated on "
+        "the InterpolationTargetTag, but it is not. See first "
+        "template parameter of std::is_same_v for the conforming "
+        "type and its templates.");
+  };
+
+  template <template <typename...> class ConformingType,
+            typename InterpolationTargetTag, typename... Ts>
+  struct test<ConformingType<InterpolationTargetTag, Ts...>> {
```

Comment:
Same comment about avoiding to "reach up" into the InterpolationTargetTag. I think you only need the temporal ID here.